src/app/events/hooks/useEventFilters.tsx

```diff
@@ -0,0 +1,146 @@
+import type { CMSCollectionName } from '@/types/cmsConfig'
+import { type NextServerSearchParams } from '@/types/searchParams'
+
+import {
+  ALL_CATEGORIES_OPTION,
+  ALL_LOCATIONS_OPTION,
+  type FilterOption,
+} from '@/constants/filterConstants'
+import { CATEGORY_KEY, LOCATION_KEY } from '@/constants/searchParams'
+
+import { getCMSFieldOptionsAndValidIds } from '@/utils/getCMSFieldOptionsAndValidIds'
+
+import type { OptionType } from '@/components/ListboxOption'
+
+import { useFilteredEntries } from '../utils/useFilteredEntries'
+
+type UseEventFiltersProps = {
+  searchParams: NextServerSearchParams
+  entries: Array<any>
+}
+
+type EventFilter = 'category' | 'location.region'
+
+type EventFilters = {
+  fields: {
+    CATEGORY: EventFilter
+    LOCATION: EventFilter
+  }
+}
+
+const EVENT_COLLECTION_NAME = 'event_entries' as CMSCollectionName
+
+const filters: EventFilters = {
+  fields: {
+    CATEGORY: 'category',
+    LOCATION: 'location.region',
+  },
+} as const
+
+export const useEventFilters = ({
```

Comment:
I haven't reviewed this fully yet - but I'm wondering if we need it.

My initial thought would be to rework `useCategory` and call it `useFilter`, so that it's a bit more flexible and we don't need to create these custom functions.

Like `useSort`, `useFilter` would take a filter function to handle custom filtering when necessary, like region for instance.

Something like that:

```
const { categoryQuery, categorizedResults } = useFilter({
    searchParams,
    entries: sortedResults,
    categoryOptions,
    filterFn: filterEventsByCategoryAndRegion
})
```

What do you think?

Reply:
Completely agree! I had in mind we would need to refactor `useCategory`, as there's an overlap with `useEventsFilter. To be honest, it felt like there were too many parts to handle in order to create 1 hook that would handle all the cases, so I created a similar one to `useCategory`, having in mind we would need consolidate the two. 

I needed fresh eyes on this code lol.